File: frontend/projects/app-support/src/app/domains/events/presence-event.ts
import { ConversationId, UserId } from '@ycyw/support-domains/chat/models'
import { PresenceStatus, Role } from '@ycyw/support-shared/enums'

export interface PresenceEvent {
  user: UserId
  role: Role
  status: PresenceStatus
  conversation: ConversationId
}

File: backend/services/support-service/support-service-application/src/main/java/com/ycyw/support/application/service/event/eventsdtos/PresenceEvent.java
package com.ycyw.support.application.service.event.eventsdtos;

import java.io.Serializable;
import java.util.UUID;

public record PresenceEvent(UUID conversationId, UUID userId, String role, Status status)
    implements Serializable {
  public enum Status {
    ONLINE("online"),
    OFFLINE("offline");

    private final String value;

    Status(String value) {
      this.value = value;
    }

    public String value() {
      return value;
    }
  }
}
